src/protocol/boolean/bit_decomposition.rs

```diff
@@ -53,32 +53,31 @@ impl BitDecomposition {
 
         // Step 5. Add back [b] bitwise. [d]_B = BitwiseSum(c, [b]_B) where d ∈ Z
         //
-        // `BitwiseSum` outputs `l + 1` bits, so [d]_B is (l + 1)-bit long.
-        let d_b = bitwise_add_constant(ctx.narrow(&Step::AddBtoC), record_id, &r.b_b, c.as_u128())
-            .await?;
+        // `BitwiseSum` outputs one more bit than its input, so [d]_B is (el + 1)-bit long.
+        let d_b = add_constant(ctx.narrow(&Step::AddBtoC), record_id, &r.b_b, c.as_u128()).await?;
 
-        // Step 6. p <=? d. The paper says "p <? d", but should actually be "p <=? d"
+        // Step 6. p <? d.
```

Comment:
I don't think you can drop the equals sign here. We are checking if the input is "greater than or equal to" the prime. If it's equal, this returns true. So I think it must remain:

```// Step 6. p <=? d```